test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -579,15 +580,21 @@ func BenchmarkPerfScheduling(b *testing.B) {
 		b.Fatal(err)
 	}
 
+	_, ctx := ktesting.NewTestContext(b)
+
```

Comment:
Move this into the `func(b *testing.B) {` below, then each test has its own output. Right now, multiple different unrelated test cases share the same output.

You may see warnings about "leaked goroutines": that is printed by the test logger when some goroutine logs something after the test has completed. It doesn't panic (like `g.Logf` normally would), so you can ignore it or use it as reminder that waiting for goroutine completion is missing.

Reply:
Updated.

Reply:
I remembered one more aspect of this: ktesting by default uses log level 5 because its main usage is unit testing where detailed logging is more important after a failure than runtime performance. But for performance benchmarks we want a lower log level. Also useful would be command line flags to increase logging and to enable JSON output. This enables testing the runtime overhead of logging.

I did all of that before, but the "contextual logging in scheduler" PR is still pending. The latest incarnation of the scheduler_perf changes are here: https://github.com/kubernetes/kubernetes/pull/111155/files#diff-3ac53aaed5c5c33f714065046108be1c9c1798fafce05276faf5dadbeb4b17f8

@kerthcet: can you perhaps include that change to test/integration/scheduler_perf/main_test.go in your PR? Then the other PR doesn't need to touch test/integration/scheduler_perf anymore at all because you are covering it completely.

Reply:
Agree, result will be more accurate in performance test with lower log level, and I'm not familiar with contextual logging, so I'll follow your advice to include the CL, but considering this is more than fixing the CI error, I'll make it an append commit then we can revert it easily if need. 

Just append the commit for better review now, once LGTM, I'll squash them with separate commits.

Reply:
Probably log level 2 would be more appropriate, as it's the recommended log level in production.

Reply:
I agree, log level 2 as default is better.

It may indeed be cleaner to take out all changes related to ktesting and submit those as a separate PR. Just replace `ctx := ktesting.NewTestContext` with `ctx := context.Background()` and the rest remains the same.

Reply:
Addressed all the comments:
1) change the log level to 2
2) squashed with different commits